picasso/src/main/java/com/squareup/picasso/Picasso.java

```diff
@@ -717,13 +732,14 @@ public Builder withCacheSize(int maxByteCount) {
       if (maxByteCount < 0) {
         throw new IllegalArgumentException("maxByteCount < 0: " + maxByteCount);
       }
-      if (cache != null) {
+      if (cacheSet) {
         throw new IllegalStateException("Memory cache already set.");
       }
+      cacheSet = true;
```

Comment:
This seems to match nullability of the field. Can we just use that instead like before?

Reply:
i can get rid of the legacy "already set" message.
but, unfortunately not this flag because i'm overloading the meaning of null.

in `build`, we have to know whether a user explicitly passed in `0` for a null/no cache or whether we have the unset value and should create the default cache for the user.